### PR TITLE
Add search REST client for ElasticSearch

### PR DESCRIPTION
**Pull request by Beacze**
- Create package search inside package elasticsearch
- Add SearchRestClient which extends KitodoRestClient
- Add SearchResult class
- Add tests

Depends on #705

### CODE REVIEW

File: Kitodo-DataManagement/src/test/java/org/kitodo/data/elasticsearch/search/SearchRestClientTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        restClient.setType("testget");
+
+        return restClient;
+    }
+
+    @BeforeClass
+    public static void prepareDatabase() throws IOException {
+        IndexRestClient indexRestClient = initializeIndexRestClient();
+        indexRestClient.addDocument(MockEntity.createEntities().get(1), 1);
+        indexRestClient.addDocument(MockEntity.createEntities().get(2), 2);
```
Comment (Beacze) [inline Kitodo-DataManagement/src/test/java/org/kitodo/data/elasticsearch/search/SearchRestClientTest.java]: Travis doesn't insert second document and it causes failed test...

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/KitodoRestClient.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+     * Close REST Client.
+     *
+     * @throws IOException
+     *             add description
+     */
+    public void closeClient() throws IOException {
+        restClient.close();
+    }
+
+    public String getIndex() {
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/KitodoRestClient.java]: Public method without JavaDoc

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/KitodoRestClient.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+     * Setter for index.
+     *
+     * @param index
+     *            - equal to the name of database, default kitodo
+     */
+    public void setIndex(String index) {
+        this.index = index;
+    }
+
+    public String getType() {
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/KitodoRestClient.java]: Public method without JavaDoc

File: Kitodo-DataManagement/src/test/java/org/kitodo/data/elasticsearch/search/SearchRestClientTest.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        SearchRestClient searchRestClient = initializeSearchRestClient();
+        String result = searchRestClient.getDocument(1);
+
+        boolean condition = result.contains("\"found\" : true");
+        assertTrue("Get of document has failed!", condition);
+    }
+
+    @Ignore("Entities are not inserted in this method...")
+    @Test
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/test/java/org/kitodo/data/elasticsearch/search/SearchRestClientTest.java]: If this test is enabled it is failing on TravisCI and my local development system.

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/KitodoRestClient.java (hunk 2 above)
Comment (Beacze) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/KitodoRestClient.java]: fixed in #705

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/KitodoRestClient.java (hunk 3 above)
Comment (Beacze) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/KitodoRestClient.java]: fixed in #705

**Pull request merged**